api/v1/common/sagemaker_api.go

```diff
@@ -528,3 +528,45 @@ type DebugRuleEvaluationStatus struct {
 
 	StatusDetail *string `json:"statusDetail,omitempty"`
 }
+
+// AutoscalingResource TODO: <link>, check validation
+type AutoscalingResource struct {
+	// +kubebuilder:validation:MinLength=1
+	EndpointName *string `json:"endpointName"`
+
+	// +kubebuilder:validation:MinLength=1
+	VariantName *string `json:"variantName"`
+}
+
+// PredefinedMetricSpecification TODO: <link>
+type PredefinedMetricSpecification struct {
+	PredefinedMetricType *string `json:"predefinedMetricType,omitempty"`
+}
+
+// CustomizedMetricSpecification TODO: <link>
+type CustomizedMetricSpecification struct {
+	MetricName *string `json:"metricName,omitempty"`
+	Namespace  *string `json:"namespace,omitempty"`
+	Statistic  *string `json:"statistic,omitempty"`
+	Unit       *string `json:"unit,omitempty"`
+	// TODO: currently reusing the existing struct but this could be changed to match the API
```

Comment:
Is this regarding Dimensions or Unit ?

Reply:
Dimension, I am reusing the AutoscalingResource struct. It should either be renamed or I should create a separate one

Reply:
Fixed.